modules/terraform-aci-contract/main.tf

```diff
@@ -54,10 +87,78 @@ resource "aci_rest_managed" "vzRsSubjFiltAtt" {
 }
 
 resource "aci_rest_managed" "vzRsSubjGraphAtt" {
-  for_each   = { for subj in var.subjects : subj.name => subj if subj.service_graph != null }
+  for_each   = { for subj in var.subjects : subj.name => subj if subj.service_graph != null && (length(subj.provider_to_consumer_filters) == 0 || length(subj.consumer_to_provider_filters) == 0) }
   dn         = "${aci_rest_managed.vzSubj[each.key].dn}/rsSubjGraphAtt"
   class_name = "vzRsSubjGraphAtt"
   content = {
     tnVnsAbsGraphName = each.value.service_graph
   }
 }
+
+resource "aci_rest_managed" "vzInTerm" {
+  depends_on = [aci_rest_managed.vzSubj]
+  for_each   = { for subj in var.subjects : subj.name => subj if length(subj.provider_to_consumer_filters) != 0 || length(subj.consumer_to_provider_filters) != 0 }
+  dn         = "${aci_rest_managed.vzBrCP.dn}/subj-${each.value.name}/intmnl"
+  class_name = "vzInTerm"
+  content = {
+    prio       = each.value.consumer_to_provider_qos_class
+    targetDscp = each.value.consumer_to_provider_target_dscp
+  }
+}
+
+resource "aci_rest_managed" "vzOutTerm" {
+  depends_on = [aci_rest_managed.vzSubj]
+  for_each   = { for subj in var.subjects : subj.name => subj if length(subj.provider_to_consumer_filters) != 0 || length(subj.consumer_to_provider_filters) != 0 }
+  dn         = "${aci_rest_managed.vzBrCP.dn}/subj-${each.value.name}/outtmnl"
+  class_name = "vzOutTerm"
+  content = {
+    prio       = each.value.provider_to_consumer_qos_class
+    targetDscp = each.value.provider_to_consumer_target_dscp
+  }
+}
+
+resource "aci_rest_managed" "vzRsFiltAtt_ctp" {
+  depends_on = [aci_rest_managed.vzInTerm]
+  for_each   = { for filter in local.subj_filter_list_ctp : filter.id => filter }
+  dn         = "${aci_rest_managed.vzSubj[each.value.subj].dn}/intmnl/rsfiltAtt-${each.value.filter}"
```

Comment:
Change `dn` to `${aci_rest_managed.vzInTerm[each.value.subj].dn}/rsfiltAtt-${each.value.filter}` and remove the `depends_on` statement.